File: BugFreeCodes/SPOJ/ToAndFro.py
"""
Author - Rohith Raj R
email - rohithrajreganti@gmail.com
Date - 2/02/2017
Problem CODE - TOANDFRO
SPOJ

"""


def chunkstring(string, length):
    return (string[0+i:length+i] for i in range(0, len(string), length))


while(True):
    
    n=int(raw_input())
    if(n==0):break
    
    x=raw_input()
    a=list(chunkstring(x,n))
    
    for i in range(0,len(a)):
        if(i%2!=0):
            a[i]=a[i][::-1]
            
    b= "".join(a)
    l1=list()
    for i in range(0,n):
        
        for k in range(i+0,i+len(x),n):
            
            l1.append(b[k])
            
    print "".join(l1)
    


    
        

File: BugFreeCodes/SPOJ/ToAndFro.c
/*

"""
Author - Rohith Raj R
email - rohithrajreganti@gmail.com
Date - 2/02/2017
Problem CODE - TOANDFRO
SPOJ

"""


*/


# include <stdio.h>
# include <stdlib.h>
# include <string.h>


int main()
{

while(1)

{

int col;
 scanf("%d",&col);
if(col==0) break;


char str[300];
scanf("%s",str);

int rows=strlen(str)/col;
int ans[rows][col];
int i,j,k=0;
for(i=0;i<rows;i++)
{

if(i%2)
{

for(j=col-1;j>=0;j--)
{

ans[i][j]=str[k++];
}


}

else
{

for(j=0;j<col;j++)

{


ans[i][j]=str[k++];

}

}


}

for(i=0;i<col;i++)
for(j=0;j<rows;j++)
printf("%c",ans[j][i]);





}



return 0 ;
}
